presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcConfig.java

```diff
@@ -99,4 +105,46 @@ public BaseJdbcConfig setUnsupportedTypeHandling(UnsupportedTypeHandling unsuppo
         this.unsupportedTypeHandling = unsupportedTypeHandling;
         return this;
     }
+
+    public boolean isUnspecifiedDecimalEnabled()
+    {
+        return unspecifiedDecimalEnabled;
+    }
+
+    @Config("unspecified-decimal-enabled")
```

Comment:
Maybe let's make them consistent:
```
decimal.enabled
decimal.default-scale
decimal.rounding-mode
```

Also I think that user should not be able to set below properties when `decimal.enabled=false`. 
```
decimal.default-scale
decimal.rounding-mode
```
See `io.prestosql.plugin.hive.security.HiveSecurityModule` and `io.airlift.configuration.ConditionalModule#installModuleIf`  how this could be achieved. 

Also we should not allow this properties to configurable for sqlserver and other jdbc based connectors that do not have such type.

Reply:
I would go with `decimal-precision-handling` instead of `decimal-enabled`.
Mapping unspecified decimals and specified exceeding decimals can lead to overflow situations due to precision issues.
With `decimal-precision-handling = strict | allow_overflow` users can decide if they want to take the risk.

Reply:
I would go with `decimal-precision-handling` instead of `decimal-enabled`.
Mapping unspecified decimals and specified exceeding decimals can lead to overflow situations due to precision issues.
```
decimal-precision-handling = strict | allow_overflow
```
With `decimal-precision-handling = allow_overflow` users can choose to take the risk of failing on decimal overflow while mapping all decimals.

With `decimal-precision-handling = strict` we will keep the current behaviour.

Reply:
How about `decimal-mapping = strict | allow_overflow `? I am not sure about `allow_overflow`, however I didn't find anything better.

The problem here is with session properties. Because we should not allow to change rounding mode or default scale when decimal mapping is disabled (strict).

Reply:
`decimal-mapping` is better since with `strict` (`strict_precision` ?)  unspecified and exceeding decimals are not mapped.
I would start by adding a statement to the description of `decimal.rounding-mode` that rounding is not relevant when strict mode is set.

Reply:
Sounds good to me.